src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy

```diff
@@ -3112,15 +3125,18 @@ public enum LegendsAwakened implements LogicCardInfo {
             energyCost ()
             attackRequirement {}
             onAttack {
-              damage 0
+              my.bench.each{
+                heal 30, it
+              }
             }
           }
           move "Supreme Blast", {
-            text "200 damage. in play, this attack does nothing. Discard all Energy attached to Mesprit."
+            text "200 damage. If you don't have Uxie LV.X and Azelf LV.X in play, this attack does nothing. Discard all Energy attached to Mesprit."
             energyCost P, P
-            attackRequirement {}
+            attackRequirement {my.all.findAll{it.name == ("Uxie Lv.X")} && my.all.findAll{it.name == ("Uxie Lv.X")}}
             onAttack {
-              damage 0
+              damage 200
+              discardAllEnergySelf(null)
```

Comment:
The method is `discardAllSelfEnergy()` and you don't need to include the `null` as it has null as the default value.